src/blib2to3/pgen2/tokenize.py

```diff
@@ -45,1070 +43,218 @@
-    **{f"{prefix}'''": single3prog_plus_lbrace for prefix in _fstring_prefixes},
-    **{f'{prefix}"""': double3prog_plus_lbrace for prefix in _fstring_prefixes},
-}
-
-triple_quoted: Final = (
-    {"'''", '"""'}
-    | {f"{prefix}'''" for prefix in _strprefixes | _fstring_prefixes}
-    | {f'{prefix}"""' for prefix in _strprefixes | _fstring_prefixes}
-)
-single_quoted: Final = (
-    {"'", '"'}
-    | {f"{prefix}'" for prefix in _strprefixes | _fstring_prefixes}
-    | {f'{prefix}"' for prefix in _strprefixes | _fstring_prefixes}
-)
-fstring_prefix: Final = tuple(
-    {f"{prefix}'" for prefix in _fstring_prefixes}
-    | {f'{prefix}"' for prefix in _fstring_prefixes}
-    | {f"{prefix}'''" for prefix in _fstring_prefixes}
-    | {f'{prefix}"""' for prefix in _fstring_prefixes}
-)
-
-tabsize = 8
-
-
-class TokenError(Exception):
-    pass
-
-
-class StopTokenizing(Exception):
-    pass
-
+del _token
 
 Coord = tuple[int, int]
+TokenInfo = tuple[int, str, Coord, Coord, str]
+
+TOKEN_TYPE_MAP = {
+    TokenType.indent: INDENT,
+    TokenType.dedent: DEDENT,
+    TokenType.newline: NEWLINE,
+    TokenType.nl: NL,
+    TokenType.comment: COMMENT,
+    TokenType.semicolon: OP,
+    TokenType.lparen: OP,
+    TokenType.rparen: OP,
+    TokenType.lbracket: OP,
+    TokenType.rbracket: OP,
+    TokenType.lbrace: OP,
+    TokenType.rbrace: OP,
+    TokenType.colon: OP,
+    TokenType.op: OP,
+    TokenType.identifier: NAME,
+    TokenType.number: NUMBER,
+    TokenType.string: STRING,
+    TokenType.fstring_start: FSTRING_START,
+    TokenType.fstring_middle: FSTRING_MIDDLE,
+    TokenType.fstring_end: FSTRING_END,
+    TokenType.endmarker: ENDMARKER,
+}
 
 
-def printtoken(
-    type: int, token: str, srow_col: Coord, erow_col: Coord, line: str
-) -> None:  # for testing
-    (srow, scol) = srow_col
-    (erow, ecol) = erow_col
-    print(
-        "%d,%d-%d,%d:\t%s\t%s" % (srow, scol, erow, ecol, tok_name[type], repr(token))
-    )
-
-
-TokenEater = Callable[[int, str, Coord, Coord, str], None]
-
-
-def tokenize(readline: Callable[[], str], tokeneater: TokenEater = printtoken) -> None:
-    """
-    The tokenize() function accepts two parameters: one representing the
-    input stream, and one providing an output mechanism for tokenize().
+class TokenError(Exception): ...
 
-    The first parameter, readline, must be a callable object which provides
-    the same interface as the readline() method of built-in file objects.
-    Each call to the function should return one line of input as a string.
 
-    The second parameter, tokeneater, must also be a callable object. It is
-    called once for each token, with five arguments, corresponding to the
-    tuples generated by generate_tokens().
+def transform_whitespace(
+    token: pytokens.Token, source: str, prev_token: Optional[pytokens.Token]
+) -> pytokens.Token:
+    r"""
+    Black treats `\\\n` at the end of a line as a 'NL' token, while it
```

Comment:
That doesn't sound particularly intentional, I'd be open to changing Black to remove this divergence.

Reply:
Feel free to! I can give you a test case with the expected behaviour.

Reply:
So, this is enough as a test case actually:

```python
a \
  b
```

But, the reason this probably exists is to support formatting [this file](https://github.com/psf/black/blob/main/tests/data/cases/backslash_before_indent.py):

```python
class Plotter:
\
    pass

class AnotherCase:
  \
    """Some
    \
    Docstring
    """
```

Reply:
I tried removing this edge case, but it just seems to break the backslash before indent test case. Let's just keep it as-is for now and try to fix it later?